documentation/incremental_settings.md

```diff
@@ -0,0 +1,85 @@
+# Incremental Settings for Ansible Values
+
+In some cases in Ansible, you provide most or all of the pertient attributes
+of the managed object.  For example, for the `file` module, you provide the
+content, permissions, ownership, and other attributes of the file to be
```

Comment:
I think this is still misleading. The `file` module actually does very fine-grained, incremental changes just as `sysctl` does. Consider the following playbook, you will see that each invocation of `file` changes only one attribute and preserves the others.
```yaml
- name: show fine grained changes with the file module
  hosts: localhost
  vars:
    fname: foo
  tasks:
    - file:
        state: touch
        name: "{{ fname }}"
    - stat:
        path: "{{ fname }}"
      register: stat0
    - name: show initial result
      debug:
        msg: "{{ stat.gr_name }} {{ stat.mode }} {{ stat.mtime }}"
      vars:
        stat: "{{ stat0.stat }}"

    - file:
        mode: g+x
        name: "{{ fname }}"
    - stat:
        path: "{{ fname }}"
      register: stat1

    - name: show first result
      debug:
        msg: "{{ stat.gr_name }} {{ stat.mode }} {{ stat.mtime }}"
      vars:
        stat: "{{ stat1.stat }}"
    - file:
        group: libvirt
        name: "{{ fname }}"
    - stat:
        path: "{{ fname }}"
      register: stat2
    - name: show second result
      debug:
        msg: "{{ stat.gr_name }} {{ stat.mode }} {{ stat.mtime }}"
      vars:
        stat: "{{ stat2.stat }}"

    - file:
        modification_time: now
        name: "{{ fname }}"
    - stat:
        path: "{{ fname }}"
      register: stat3
    - name: show third result
      debug:
        msg: "{{ stat.gr_name }} {{ stat.mode }} {{ stat.mtime }}"
      vars:
        stat: "{{ stat3.stat }}"

    - file:
        state: absent
        name: "{{ fname }}"
```
If you need an example of an Ansible module that replaces the whole object, `template` or `copy` modules might be the best. I believe that if you go through the core modules, you will find though that incremental, fine-grained changes and preservation of unspecified values is actually the usual behavior for Ansible modules and replacing the whole managed object is more an exception.
I propose to change the wording as follows:
"Ansible playbooks and roles are often written so that they declare the complete state of the subsystem that they manage and entirely replace the previous state, such as the content of the managed configuration files. On the other hand, Ansible modules usually have an interface where you have to specify ..."

Reply:
Fixed in https://github.com/linux-system-roles/linux-system-roles.github.io/pull/21/commits/373b5ac990e225b3ac11431fef1154a65e047475
Note that I got rid of the text to try to explain the difference between managing the whole state of the system or of the managed objects, and instead explained it as generalizing the concept of incremental changes as implemented by modules, and using that approach for system roles.